app/scripts/controllers/network/network.js

```diff
@@ -120,6 +135,42 @@ export default class NetworkController extends EventEmitter {
     return { provider, blockTracker };
   }
 
+  /**
+   * Method to return the latest block for the current network
+   * @returns {Object} Block header
+   */
+  getLatestBlock() {
+    return new Promise((resolve, reject) => {
+      const { provider } = this.getProviderAndBlockTracker();
+      const ethQuery = new EthQuery(provider);
+      ethQuery.sendAsync(
+        { method: 'eth_getBlockByNumber', params: ['latest', false] },
+        (err, block) => {
+          if (err) {
+            return reject(err);
+          }
+          return resolve(block);
+        },
+      );
+    });
+  }
+
+  /**
+   * Method to check if the block header contains fields that indicate EIP 1559
+   * support (baseFeePerGas).
+   * @returns {boolean} true if current network supports EIP 1559
+   */
+  async getEIP1559Compatibility() {
```

Comment:
Wondering if, since you've made the networkDetails store quite general to all EIPs, whether we should have a method that is abstracted away from the particular EIP we're checking against. Obviously this is the important one right now, so not a big deal, just a thought given the open-endedness of the networkDetails store shape.

Reply:
I think we'd need specific implementations for each one, this function is checking for a specific field on a block header...other EIPs will have other detection methods, there is a generic method for *saving* support for an EIP and one for clearing all support.

Reply:
yeah makes sense. It wouldn't make any sense until there are more EIPs to check against, but I was thinking something more like a reducer with cases for different EIPs

Reply:
But definitely not a REQ here.

Reply:
Yeah we'd need to think that one through a bit, but not opposed to it long term.